pkg/registry/manifest.go

```diff
@@ -193,3 +211,59 @@ func (m *manifests) handle(resp http.ResponseWriter, req *http.Request) *regErro
 		Message: "We don't understand your method + url",
 	}
 }
+
+func (m *manifests) handleTags(resp http.ResponseWriter, req *http.Request) *regError {
+	elem := strings.Split(req.URL.Path, "/")
+	elem = elem[1:]
+	repo := strings.Join(elem[1:len(elem)-2], "/")
+	query := req.URL.Query()
+	nStr := query.Get("n")
+	n := 1000
+	if nStr != "" {
+		n, _ = strconv.Atoi(nStr)
+	}
+
+	if req.Method == "GET" {
+		m.lock.Lock()
+		defer m.lock.Unlock()
+
+		c, ok := m.manifests[repo]
+		if !ok {
+			return &regError{
+				Status:  http.StatusNotFound,
+				Code:    "NAME_UNKNOWN",
+				Message: "Unknown name",
+			}
+		}
+
+		var tags []string
+		countTags := 0
+		for tag := range c {
+			if countTags >= n {
+				break
```

Comment:
I'd be satisfied with just a TODO here that we need to include the `Link` header for the next page, but you could also just implement it, if you'd like!

https://github.com/opencontainers/distribution-spec/blob/b505e9cc53ec499edbd9c1be32298388921bb705/detail.md#tags-paginated

Reply:
@jonjohnsonjr thanks for the feedback, i will implement that

Reply:
@jonjohnsonjr currently the crane does not support pagination, this is something we would like to have as input flags in the command?

Reply:
I tried to run the list tags against docker API and I think I'm doing something wrong because when I set the `n` to be something like 5 or 100 it returns the same as not setting the `n`

```
TOKEN=$(curl -s "https://auth.docker.io/token?scope=repository%3Alibrary%2Fgolang%3Apull&service=registry.docker.io" | jq -r .token)
curl -v -s -H "Authorization: Bearer ${TOKEN}" "https://index.docker.io/v2/library/golang/tags/list?n=5"
```

am I doing something very wrong?

Reply:
tried in gcr as well same results :/

Reply:
> am I doing something very wrong?

Nope! It's just that neither docker hub nor GCR implement this (for different reasons).

Looking for a random public image on the internet, it seems that quay.io does implement this:

```
$ curl -v https://quay.io/v2/kubernetes_incubator/nfs-provisioner/tags/list?n=5
```

Reply:
@jonjohnsonjr I added a TODO, because I'm afraid I cannot implement this due to lack of time and would not have this PR hanging around for a long time.

When I have more time I will comeback if this was not implemented by another person.

PTAL

Reply:
Sounds good, thanks!